examples/server.rs

```diff
@@ -66,14 +70,33 @@ async fn run_server() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
             .boxed();
 
     let service = make_service_fn(|_| async { Ok::<_, io::Error>(service_fn(echo)) });
-    let server = Server::builder(incoming_tls_stream).serve(service);
+    let server = Server::builder(HyperAcceptor {
+        acceptor: incoming_tls_stream,
+    })
+    .serve(service);
 
     // Run the future, keep going until an error occurs.
     println!("Starting to serve on https://{}.", addr);
     server.await?;
     Ok(())
 }
 
+struct HyperAcceptor {
+    acceptor: Pin<Box<dyn Stream<Item = Result<TlsStream<TcpStream>, io::Error>>>>,
+}
+
+impl hyper::server::accept::Accept for HyperAcceptor {
+    type Conn = TlsStream<TcpStream>;
+    type Error = io::Error;
+
+    fn poll_accept(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context,
+    ) -> Poll<Option<Result<Self::Conn, Self::Error>>> {
+        Pin::new(&mut self.acceptor).poll_next(cx)
+    }
+}
+
```

Comment:
I'm not too sure if this is the best solution. Basically hyper doesn't really use Stream anymore, so they have alternative traits such as Accept that you need to implement. Maybe something like this should be in hyper-rustls directly and not in an example?